dask/array/core.py

```diff
@@ -1843,7 +1846,8 @@ def normalize_chunks(chunks, shape=None):
                 "Got chunks=%s, shape=%s" % (chunks, shape))
 
     if shape is not None:
-        chunks = tuple(c if c is not None else s for c, s in zip(chunks, shape))
+        chunks = tuple(c if c not in {None, -1} else s
```

Comment:
I also considered using `-1` to indicate "the remaining available size" but the logic there was slightly trickier (and I'm not sure anyone would actually use it).

Reply:
Yeah, I think the current behavior is simple and probably covers the common case.